### PR TITLE
Shuffle init procs

### PR DESCRIPTION
**Pull request by adacovsk**
I looked at World/New and found a bunch of inits that didn't belong there that weren't related to database stuff... Which isn't good for diagnosing potential issues. Ideally, all such things would be in their own subsystem or "hooks" as N3x15 would have wanted. Maybe slowly, I'll work on putting these things in their own subsystem? The hook stuff was never finished and subsystems work well, at least for initialization stuff.

## What this does
- moves some init procs to master (where they don't belong, but they REALLY don't belong in world/new())
- moves some init procs to the "more_init" umbrella subsystem for orphaned and alone inits
- ~~removes a one second sleep~~ can't give a good reason why other than "it just works"
- refactors holiday.dm a little
- Slightly modifies lighting init so that players aren't left in the dark for several seconds before the lights turn on. This got worse with higher player count. Instead, lights start on, then are turned off based on empty departments.

## Why it's good
- organisation

### CODE REVIEW

File: code/controllers/mc/master.dm [hunk 1]
```diff
@@ -124,15 +124,21 @@ var/CURRENT_TICKLIMIT = TICK_LIMIT_RUNNING
 // 	Make a subsystem, give it the SS_NO_FIRE flag, and do your work in it's Initialize()
 /datum/controller/master/proc/Setup()
 	set waitfor = FALSE
-	sleep(1 SECONDS)
-	to_chat(world, "<span class='boldannounce'>Initializing subsystems...</span>")
-
+	sleep(-1)
+	//moving this random bullshit into here, because it didn't belong in world/New()
+	//SetupHooks() // /N3X15 project from 8 years ago (WIP)
+	createDatacore()
```
Comment (DamianX) [inline code/controllers/mc/master.dm:130]: why are these procs procs? Why do they return one? Why not put the contents of those procs here directly?

File: code/controllers/mc/master.dm [hunk 2]
```diff
@@ -124,15 +124,21 @@ var/CURRENT_TICKLIMIT = TICK_LIMIT_RUNNING
 // 	Make a subsystem, give it the SS_NO_FIRE flag, and do your work in it's Initialize()
 /datum/controller/master/proc/Setup()
 	set waitfor = FALSE
-	sleep(1 SECONDS)
-	to_chat(world, "<span class='boldannounce'>Initializing subsystems...</span>")
-
+	sleep(-1)
```
Comment (DamianX) [inline code/controllers/mc/master.dm]: Why did you replace this?

File: code/controllers/mc/master.dm [hunk 3]
```diff
@@ -124,15 +124,21 @@ var/CURRENT_TICKLIMIT = TICK_LIMIT_RUNNING
...
+	//moving this random bullshit into here, because it didn't belong in world/New()
```
Comment (DamianX) [inline code/controllers/mc/master.dm:128]: How did you determine these things didn't belong in `world/New()`?

File: code/controllers/mc/master.dm [hunk 4]
```diff
@@ -124,15 +124,21 @@ var/CURRENT_TICKLIMIT = TICK_LIMIT_RUNNING
...
+	//SetupHooks() // /N3X15 project from 8 years ago (WIP)
```
Comment (DamianX) [inline code/controllers/mc/master.dm:129]: A quick search indicates there are 7 `/hook_handler`s in use and 18 `/hook`s defined, yet you decided to disable them. Why?

File: code/game/gamemodes/events/holidays/Holidays.dm [hunk 5]
```diff
@@ -1,26 +1,15 @@
-//Uncommenting ALLOW_HOLIDAYS in config.txt will enable Holidays
```
Comment (DamianX) [inline code/game/gamemodes/events/holidays/Holidays.dm:1]: This comment seems relevant, why did you remove it?

File: code/game/gamemodes/events/holidays/Holidays.dm [hunk 6]
```diff
@@ -1,26 +1,15 @@
...
-var/global/Holiday = null
-
-//Just thinking ahead! Here's the foundations to a more robust Holiday event system.
-//It's easy as hell to add stuff. Just set Holiday to something using the switch (or something else)
```
Comment (DamianX) [inline code/game/gamemodes/events/holidays/Holidays.dm:5]: Why did you remove these instructions?

File: code/controllers/mc/master.dm (hunk 1 above)
Comment (adacovsk) [reply code/controllers/mc/master.dm:130]: I'd plan to put them in a subsystem eventually, much is likely more appropriate, then that way they can be removed and the SS init will handle them properly

File: code/controllers/mc/master.dm (hunk 2 above)
Comment (adacovsk) [reply code/controllers/mc/master.dm]: sleep(-1) works much better in this case

Comment (DamianX) [reply code/controllers/mc/master.dm]: How so?

File: code/controllers/mc/master.dm (hunk 3 above)
Comment (adacovsk) [reply code/controllers/mc/master.dm:128]: world/New contains stuff related to metadata and database stuff, whereas the other stuff that I moved into master is for in-game stuff, which should be in a subsystem with high priority. However, that would be a big undertaking. As for the other stuff that I moved from master, into more_init, they are low priority

File: code/controllers/mc/master.dm (hunk 4 above)
Comment (adacovsk) [reply code/controllers/mc/master.dm:129]: N3X15 said he didn't test it and didn't get it working but I did miss that... I can undo it

File: code/game/gamemodes/events/holidays/Holidays.dm (hunk 6 above)
Comment (adacovsk) [reply code/game/gamemodes/events/holidays/Holidays.dm:5]: I guess I'll add my own instructions, I simplified the code so it just returns the holiday using GetHoliday() rather than assigning it directly to the Holiday global

File: code/controllers/subsystem/init/more_init_stuff.dm [hunk 7]
```diff
@@ -9,8 +9,16 @@ var/datum/subsystem/more_init/SSmore_init
 	NEW_SS_GLOBAL(SSmore_init)
 
 /datum/subsystem/more_init/Initialize(timeofday)
-	setup_news()
 	var/watch=start_watch()
+	log_startup_progress("Initializing stuff formerly left in world/New...")
+	initialize_rune_words()
+	library_catalog.initialize()
+	initialize_beespecies()
```
Comment (DeityLink) [inline code/controllers/subsystem/init/more_init_stuff.dm:16]: and that's what broke mapped chill bee packets
